packages/driver/src/cypress/location.coffee

```diff
@@ -214,8 +216,9 @@ class $Location
         memo.push _.trim(segment, "/")
       memo
     , [_.trimEnd(from, "/")]
-
-    paths.join("/")
+    ## If there is a query string and last is an empty string, don't append a /
+    if not last.includes("/") and paths.some((path) -> reQueryParam.test(path)) then paths.join("")
```

Comment:
To make this easier to read, refactor this from:

```
if ... then ...
else ...
```
to:

```
if ...
    return ...

return ...

Reply:
Is the last return necessary? Or is it just there for stylistic reasons?

Reply:
No, it's not necessary, coffeescript has implicit returns. You could go without it here